tutorials/inputs/controllers_gamepads_joysticks.rst

```diff
@@ -0,0 +1,246 @@
+be equal to ``0.0`` due to how the controller is physically built. Instead, it
+will linger at a low value such as ``0.062``. This phenomenon is known as
+*drifting* and can be more noticeable on old or faulty controllers.
+
+Let's take a racing game as a real-world example. Thanks to analog inputs, we
+can steer the car slowly in one direction or another. However, without a
+deadzone system, the car would slowly steer by itself even if the player isn't
+touching the joystick. This is because the directional axis strength won't be
+equal to ``0.0`` when we expect it to. Since we don't want our car to steer by
+itself in this case, we define a "dead zone" value of ``0.2`` which will ignore
+all input whose strength is lower than ``0.2``. An ideal dead zone value is high
+enough to ignore the input caused by joystick drifting, but is low enough to not
+ignore actual input from the player.
+
+Godot features a built-in dead zone system to tackle this problem. The default
+value is ``0.2``, but you can increase it or decrease it on a per-action basis
+in the Project Settings' Input Map tab.
+For ``Input.get_vector()``, the deadzone can be specified, or otherwise it
+will calculate the average deadzone value from all of the actions in the vector.
+
+"Echo" events
+^^^^^^^^^^^^^
+
+Unlike keyboard input, holding down a controller button such as a D-pad
+direction will **not** generate repeated input events at fixed intervals (also
+known as "echo" events). This is because the operating system never sends "echo"
+events for controller input in the first place.
+
+If you want controller buttons to send echo events, you will have to generate
+:ref:`class_InputEvent` objects by code and parse them using
+:ref:`Input.parse_input_event() <class_Input_method_parse_input_event>`
+at regular intervals. This can be accomplished
+with the help of a :ref:`class_Timer` node.
+
+Troubleshooting
+---------------
+
+.. seealso::
+
+    You can view a list of
+    `known issues with controller support <https://github.com/godotengine/godot/issues?q=is%3Aopen+is%3Aissue+label%3Atopic%3Ainput+gamepad>`__
+    on GitHub.
+
+My controller isn't recognized by Godot.
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+First, check that your controller is recognized by other applications. You can
+use the `Gamepad Tester <https://gamepad-tester.com/>`__ website to confirm that
+your controller is recognized.
+
+My controller has incorrectly mapped buttons or axes.
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If buttons are incorrectly mapped, this may be due to an erroneous mapping from
+the `SDL game controller database <https://github.com/gabomdq/SDL_GameControllerDB>`__.
+You can contribute an updated mapping to be included in the next Godot version
+by opening a pull request on the linked repository.
+
+There are many ways to create mappings. One option is to use the mapping wizard
+in the `official Joypads demo <https://godotengine.org/asset-library/asset/140>`__.
```

Comment:
Note: This is technically not true just yet, but https://github.com/godotengine/godot-demo-projects/pull/575 was merged, so this sentence will be true the next time I make a release of the demos on the Godot Asset Library.